### PR TITLE
[recipe, doc] feat: add Qwen3-Omni MMK12 NPU recipe

### PR DESCRIPTION
**Pull request by Sky-Trigger**
### What does this PR do?

This PR adds a dedicated Ascend NPU launch script for Qwen3-Omni Thinker GSPO
LoRA training on MMK12 and documents how to use it.

The NPU recipe follows the existing MMK12 V1 configuration while adding the
runtime settings needed by the validated Ascend setup:

- Sets `VLLM_ASCEND_ENABLE_NZ=0` for the vLLM Ascend rollout.
- Uses a sparse `cudagraph_capture_sizes` list because capturing too many graph
  shapes can cause runtime errors.
- Makes the MMK12 reward function path configurable through
  `REWARD_FUNCTION_PATH`, while preserving the existing scorer as the default.
- Configures the recipe for 16 NPUs and TensorBoard logging.
- Documents the NPU launch command and the graph-capture consideration in the
  GSPO trainer README.

This is an example and documentation change. It does not modify trainer,
rollout, model, or reward-scoring implementation code.

### Checklist Before Starting

- [x] Search for similar PRs. No matching open PR was found for
  [MMK12 NPU](https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+MMK12+NPU)
  or
  [Qwen3-Omni Ascend](https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+Qwen3-Omni+Ascend).
- [x] Format the PR title as `[{modules}] {type}: {description}`. The proposed
  title passes `tests/special_sanity/check_pr_title.py`.

### Test

The following static checks passed:

```bash
bash -n examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_v1_npu.sh
git diff --check origin/main...HEAD
PR_TITLE='[recipe, doc] feat: add Qwen3-Omni MMK12 NPU recipe' \
  python3 tests/special_sanity/check_pr_title.py
```

A focused pre-commit run passed the documentation, docstring, license, device
API, DataProto, test-structure, naming, and compile-all hooks. The global
`autogen-trainer-cfg` hook could not complete because the current Python
environment does not provide `omegaconf`; no generated configuration is changed
by this PR.

End-to-end NPU training results will be added before requesting review.

#### NPU training results

439618b6364b10274f468538fa039d3e

Experiment configuration: `<model / NPU count / batch size / training steps / rollout settings>`

Observation: `<summarize reward behavior, training stability, and other key metrics>`

### API and Usage Example

Launch the Ascend NPU recipe with the default MMK12 reward scorer:

```bash
TRAIN_FILE=$HOME/data/mmk12/train.parquet \
VAL_FILE=$HOME/data/mmk12/test.parquet \
bash examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_v1_npu.sh
```

The model, dataset, and reward scorer paths can be overridden without editing
the script:

```bash
MODEL_PATH=/path/to/Qwen3-Omni-30B-A3B-Instruct \
TRAIN_FILE=/path/to/train.parquet \
VAL_FILE=/path/to/test.parquet \
REWARD_FUNCTION_PATH=/path/to/custom_reward.py \
bash examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_v1_npu.sh
```

Additional Hydra overrides can still be appended to the command.

#### NPU recipe

- Adds
  `examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_v1_npu.sh`.
- Keeps the existing MMK12 preprocessing and `compute_score` reward contract.
- Adds Ascend rollout environment and graph-capture settings without changing
  the GPU recipe.

#### Documentation

- Adds the NPU launch command to `examples/gspo_trainer/README.md`.
- Explains why the capture-size list is intentionally sparse.

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]
LGTM
